plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/LdapGroupProviderConfig.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.password.ldap;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.ConfigSecuritySensitive;
+import io.airlift.units.Duration;
+
+import javax.validation.constraints.NotNull;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Objects.requireNonNull;
+
+public class LdapGroupProviderConfig
+{
+    private List<String> userBindSearchPatterns = ImmutableList.of();
+    private String groupAuthorizationSearchPattern;
+    private String userBaseDistinguishedName;
+    private String groupBaseDistinguishedName;
+    private String bindDistinguishedName;
+    private String bindPassword;
+    private boolean allowUserNotExist;
+    private String groupSearchFilter;
+    private Duration ldapCacheTtl = new Duration(1, TimeUnit.HOURS);
+
+    @NotNull
```

Comment:
Can we apply additional assertion on `NotEmpty`

Reply:
yup, adding ..